### ISSUE TITLE
OutOfMemory when creating a simulation

### ISSUE THREAD

**Issue by PavelBal** [Open-Systems-Pharmacology/MoBi#109]
I cannot create a simulation from my building blocks - an error is thrown

Application:
MoBi 7.1.0 - Build 66

Exception of type 'System.OutOfMemoryException' was thrown.

Stack trace:
```
at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at OSPSuite.Core.Services.CircularReferenceChecker.buildEntityReferenceCache(IUsingFormula usingFormula)
   at OSPSuite.Utility.Extensions.EnumerableExtensions.Each[T](IEnumerable`1 list, Action`1 action)
   at OSPSuite.Core.Services.CircularReferenceChecker.CheckCircularReferencesIn(IModel model, IBuildConfiguration buildConfiguration)
   at OSPSuite.Core.Domain.Services.ModelConstructor.checkCircularReferences(IModel model, IBuildConfiguration buildConfiguration)
   at OSPSuite.Core.Domain.Services.ModelConstructor.validateModel(IModel model, IBuildConfiguration buildConfiguration)
   at OSPSuite.Core.Domain.Services.ModelConstructor.CreateModelFrom(IBuildConfiguration buildConfiguration, String modelName)
   at MoBi.Presentation.Presenter.CreateSimulationPresenter.createModel()
   at MoBi.Presentation.Presenter.CreateSimulationPresenter.<>c__DisplayClass17_0.<finish>b__0()
   at OSPSuite.Presentation.Services.HeavyWorkManager.doWork()
   at OSPSuite.Presentation.Services.HeavyWorkManager.<.ctor>b__9_1(Object o, DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
```

This was possible with 7.0.

I cannot share the project on GitHub.

**Comment by PavelBal** [comment]
I think this one can be closed as soon x64 will be default.

**Comment by Yuri05** [comment]
should be solved with 7.4
